frontend/public/components/secrets/create-secret/EditSecret.tsx

```diff
@@ -1,81 +1,42 @@
-import * as _ from 'lodash-es';
 import * as React from 'react';
-import { useTranslation, withTranslation } from 'react-i18next';
-import { WithT } from 'i18next';
+import { useTranslation } from 'react-i18next';
 import { useParams } from 'react-router-dom-v5-compat';
-import { LoadingBox } from '@console/shared/src/components/loading/LoadingBox';
 import { StatusBox } from '@console/shared/src/components/status/StatusBox';
-import { K8sResourceKind } from '../../../module/k8s';
-import { Firehose } from '../../utils/firehose';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { SecretModel } from '@console/internal/models';
 import { SecretTypeAbstraction } from './types';
 import { toTypeAbstraction } from './utils';
 import { SecretFormWrapper } from './SecretFormWrapper';
 
-export const SecretLoadingWrapper = withTranslation()(
-  class SecretLoadingWrapper extends React.Component<
-    SecretLoadingWrapperProps & WithT,
-    SecretLoadingWrapperState
-  > {
-    readonly state: SecretLoadingWrapperState = {
-      secretTypeAbstraction: SecretTypeAbstraction.generic,
-    };
-    componentDidUpdate() {
-      if (!_.isEmpty(this.props.obj.data)) {
-        const secretTypeAbstraction = toTypeAbstraction(this.props.obj.data);
-        if (this.state.secretTypeAbstraction !== secretTypeAbstraction) {
-          this.setState({
-            secretTypeAbstraction,
-          });
-        }
-      }
-    }
-    render() {
-      const { obj, fixedKeys } = this.props;
-      const { secretTypeAbstraction } = this.state;
-      if (!secretTypeAbstraction) {
-        return <LoadingBox />;
-      }
-      const fixed = fixedKeys?.reduce((acc, k) => ({ ...acc, [k]: obj.data?.[k] || '' }), {});
+export const EditSecret: React.FC = () => {
+  const params = useParams();
```

Comment:
nit, use destructuring for consistency
```suggestion
  const { name, ns } = useParams();
```